src/v2/components/PotentialFireList.jsx

```diff
@@ -113,19 +110,24 @@ export default function PotentialFireList() {
   }, [includesAllFires, updateFires])
 
   useEffect(() => {
-    eventSourceRef.current = getEventSource('/fireEvents')
+    if (eventSourceRef.current == null) {
+      eventSourceRef.current = getEventSource('/fireEvents')
+    }
 
     const {current: eventSource} = eventSourceRef
 
+    function handleClosedConnection() {
+      eventSourceRef.current = null
+      tidy()
+    }
+
     function tidy() {
       eventSource.removeEventListener('newPotentialFire', handlePotentialFire)
-      eventSource.removeEventListener('closedConnection', tidy)
-      eventSource.close() // Does nothing if the connection is already closed.
-      eventSourceRef.current = null
+      eventSource.removeEventListener('closedConnection', handleClosedConnection)
     }
 
     eventSource.addEventListener('newPotentialFire', handlePotentialFire)
-    eventSource.addEventListener('closedConnection', tidy)
+    eventSource.addEventListener('closedConnection', handleClosedConnection)
 
     return tidy
```

Comment:
Should this return tidy or handleClosedConnection?

Reply:
`tidy()`. The issue is that whatever function is returned from the `useEffect()` callback will be used to clean up when the callback is called again (i.e., each time that `handlePotentialFires` changes). If we were to call `handleClosedConnection()` instead of `tidy()`, these code changes wouldn’t have any effect—`eventSourceRef.current` would be set to `null` and the subsequent effect would (re)initialize a new event source every time.